### PR TITLE
feat(pipeline): pipeline render extension image for different architecture

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
pipeline render extension image for different architecture

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    Support pipeline render extension image for different architecture          |
| 🇨🇳 中文    |    Support pipeline render extension image for different architecture          |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: apistructs/cluster_info.go [hunk 1]
```diff
@@ -100,6 +100,14 @@ func (info ClusterInfoData) Get(key ClusterInfoMapKey) string {
 	return info[key]
 }
 
+func (info ClusterInfoData) ToMap() map[string]string {
+	dat := make(map[string]string)
```
Comment (sfwn) [inline apistructs/cluster_info.go:104]: create with capacity

File: internal/tools/pipeline/providers/reconciler/taskrun/taskop/prepare.go [hunk 2]
```diff
@@ -192,7 +192,8 @@ func (pre *prepare) makeTaskRun() (needRetry bool, err error) {
 	extSearchReq = append(extSearchReq, getActionAgentTypeVersion())
 	extSearchReq = append(extSearchReq, pre.ActionMgr.MakeActionTypeVersion(&task.Extra.Action))
 	actionDiceYmlJobMap, actionSpecYmlJobMap, err := pre.ActionMgr.SearchActions(extSearchReq,
-		actionmgr.SearchOpWithRender(map[string]string{"storageMountPoint": mountPoint}))
+		actionmgr.SearchOpWithRender(map[string]string{"storageMountPoint": mountPoint}),
+		actionmgr.SearchOpWithClusterInfo(clusterInfo.ToMap()))
```
Comment (sfwn) [inline internal/tools/pipeline/providers/reconciler/taskrun/taskop/prepare.go:196]: Could the cluterinfo be empty?

File: internal/tools/pipeline/providers/actionmgr/impl.go [hunk 3]
```diff
@@ -61,7 +61,13 @@ func (s *provider) SearchActions(items []string, ops ...OpOption) (map[string]*d
 			}
 			diceYmlStr = string(rendered)
 		}
-		diceYml, err := diceyml.New([]byte(diceYmlStr), false)
+		var diceYml *diceyml.DiceYaml
+		var err error
+		if so.ClusterInfo != nil {
+			diceYml, err = diceyml.New([]byte(diceYmlStr), false, diceyml.WithPlatformInfo(so.ClusterInfo))
```
Comment (sfwn) [inline internal/tools/pipeline/providers/actionmgr/impl.go:67]: how about pass nil map to diceyml parser?

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/tools/pipeline/providers/actionmgr/impl.go (hunk 3 above)
Comment (chengjoey) [reply internal/tools/pipeline/providers/actionmgr/impl.go:67]: if pass nil map to diceyml and with platform visitor, `diceyml.New` will return error about image replaced-failure.


`SearchActions` was called in different funcs, only `taskop.Prepare` need replace image placeholder

**Review by sfwn — DISMISSED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**